pkg/crc/cluster/cluster.go

```diff
@@ -291,23 +313,15 @@ func EnsurePullSecretPresentOnInstanceDisk(sshRunner *ssh.Runner, pullSecret *Pu
 	return sshRunner.CopyData([]byte(content), vmPullSecretPath, 0600)
 }
 
-func WaitforRequestHeaderClientCaFile(ocConfig oc.Config) error {
-	if err := WaitForOpenshiftResource(ocConfig, "configmaps"); err != nil {
-		return err
-	}
-
+func WaitforRequestHeaderClientCaFile(sshRunner *ssh.Runner) error {
 	lookupRequestHeaderClientCa := func() error {
-		cmdArgs := []string{"get", "configmaps/extension-apiserver-authentication", `-ojsonpath={.data.requestheader-client-ca-file}`,
-			"-n", "kube-system"}
-
-		stdout, stderr, err := ocConfig.RunOcCommand(cmdArgs...)
+		expired, err := CheckAggregatorClientCAValidity(sshRunner)
 		if err != nil {
-			return fmt.Errorf("Failed to get request header client ca file %v: %s", err, stderr)
+			return fmt.Errorf("Failed to expiry date: %v", err)
 		}
-		if stdout == "" {
-			return &errors.RetriableError{Err: fmt.Errorf("missing .data.requestheader-client-ca-file")}
+		if expired {
+			return &errors.RetriableError{Err: fmt.Errorf("certificate still expired")}
```

Comment:
Were you able to test this part of the PR? This PR is again doing 2 things at once, it checks certificate validity before doing the `waitForRequestHeaderClientCaFile`/`deletePods` dance, which is good for startup speed. But it also changes the implementation of `waitForRequestHeaderClientCaFile`, and this is not immediately obvious from the commit log. Imo we need to be careful with this part of the change as this workaround has been troublesome in the past. Maybe split it in a second commit?

Reply:
I tested it with an old bundle (4.5.4 from August) and the last one (4.5.14). I will split it 2 commits!